### ISSUE TITLE
Feature Request: Open in Container from Bookmark

### ISSUE THREAD

**Issue by utahcon** [mozilla/multi-account-containers#1714]
I LOVE Firefox Containers. They make my life as a developer so much easier. However, I would love to see the Bookmark interface updated to allow a bookmark to be opened in a container type.

**Comment by adnichols** [comment]
Agree with this - even the ability to add some metadata to the bookmark which would hint at what container to use would be super handy.

**Comment by kuzi-moto** [comment]
This feature would be super useful. I'm surprised it's only been requested recently.

**Comment by Solid-Ice8** [comment]
There's already an existing ad-on for this as well.

"Open bookmark in Container Tab menu item".

**Comment by adnichols** [comment]
@Solid-Ice8 that's not what is being requested. There are already ways to open a bookmark by right clicking the bookmark and selecting the container to open it in - this add-on actually is redundant with functionality already provided by the multi-account-containers add-on. Neither of these add-on features persist that selection so that just opening the bookmark results in the right container being used - you always have to right click & navigate menus. We have bookmarks which are always opened in the same container, but are not distinctive enough (for example, different google documents) to be supported by the "Always open this site in container" option.

**Comment by utahcon** [comment]
As @adnichols said the add-on @Solid-Ice8 suggested is only half a solution. Thanks for pointing that out @Solid-Ice8. However, yes, having this as a permanent solution would be preferred.

**Comment by Solid-Ice8** [comment]
No problem, I understand. However, I don't believe of any Dev's have been around for a long time.

I've seen (& been seeing) many repeated features/suggestions/issues (including both of my suggestions).

Some even go as far back as issue number 600's..and now we're in the 1700's...

**Comment by rob-miller** [comment]
The use case which depends on this feature is the ability to have multiple gmail accounts in different bookmarks.  In my hands they all resolve to https://mail.google.com/mail/u/0/#inbox so there is no way to differentiate the container based on URL alone.

> this add-on actually is redundant with functionality already provided by the multi-account-containers add-on

I have the multi-containers add-on installsed, without the additional add-on referenced by @Solid-Ice8 there is no 'open in container' option on any of my bookmarks, only on links in loaded pages.

**Comment by Solid-Ice8** [comment]
That's good that you've found it.

The suggested add-on actually displays the colours of container tabs instead MAC's built-in one. If you noticed, the container tab colours are all black in the built-in context menu. This may be helpful, but, it's not exactly perfect.

Additionally, TST (Tree Style Tabs) users like myself like to tweak around with the CSS (cascade style sheet) to customize its look. Some people enabled additional CSS only displaying the FavIcon (you know, the little icon next to websites) and not the text.

When I used this, I noticed it applies to all bookmarks too, not just the sidebar. Not all websites have their own FavIcons. 

I'm pretty sure it'll also affect MAC's built-in bookmark functionality as well. I haven't tested this yet, can't say if this is actually the case.

I'm now back to work so, I won't be able to do tests as often.
